### PR TITLE
chore(ui): migrate Button component to TS

### PR DESCRIPTION
**Pull request by gjaskiewicz**
# Summary

Migrate button component to TypeScript

# Changes Made

- migrate Button component to TS
- migrate ButtonRow component to TS

# Related Issues

- Issue 1: [link to issue](https://github.com/cloudoperators/juno/issues/217)

# Testing Instructions

1. `npm i`
2. `turbo test`

# Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have made corresponding changes to the documentation (if applicable).
- [ ] My changes generate no new warnings or errors.

### CODE REVIEW

File: packages/ui-components/src/components/Button/Button.test.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    const onClickSpy = vi.fn()
+    render(<Button onClick={onClickSpy} />)
+    screen.getByRole("button").click()
+    expect(onClickSpy).toHaveBeenCalled()
+  })
+
+  test("onclick handler is not called when disabled", () => {
+    const onClickSpy = vi.fn()
+    render(<Button disabled onClick={onClickSpy} />)
+    screen.getByRole("button").click()
```
Comment (barsukov) [inline packages/ui-components/src/components/Button/Button.test.tsx:47]: Use here act please

File: packages/ui-components/src/components/Button/Button.test.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  test("renders an <a> element with role button if href is passed", () => {
+    render(<Button href="http://blah.com" />)
+    expect(screen.getByRole("button")).toBeInTheDocument()
+    expect(screen.getByRole("button")).toHaveAttribute("href")
+  })
+
+  test("onclick handler is called as passed", () => {
+    const onClickSpy = vi.fn()
+    render(<Button onClick={onClickSpy} />)
+    screen.getByRole("button").click()
```
Comment (barsukov) [inline packages/ui-components/src/components/Button/Button.test.tsx:40]: Use here act please

File: packages/ui-components/src/components/ContentHeading/ContentHeading.stories.tsx [hunk 3]
```diff
@@ -5,7 +5,7 @@
 
 import React from "react"
 import { ContentHeading, ContentHeadingProps } from "./index"
-// import { Button } from "../Button/index.js"
+// import { Button } from "../../deprecated_js/Button/index.js"
```
Comment (barsukov) [inline packages/ui-components/src/components/ContentHeading/ContentHeading.stories.tsx:8]: please delete it

File: packages/ui-components/src/deprecated_js/Button/button.scss [hunk 4]
```diff
@@ -0,0 +1,73 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Juno contributors
+// SPDX-License-Identifier: Apache-2.0
+
+// handle button colours here, not in tw because of overall complexity, mostly singular colors, gradients, etc; and conflicting bg-colors and bg-images:
+
+
+.juno-button-default {
+	background-color: var(--color-button-default-bg);
```
Comment (barsukov) [inline packages/ui-components/src/deprecated_js/Button/button.scss:8]: I think again you have a format issue

**Review by barsukov — COMMENTED** [review]
Please fix all comments, I leave them as a comments cause going to leave for few days :)

File: packages/ui-components/src/components/ContentHeading/ContentHeading.stories.tsx (hunk 3 above)
Comment (gjaskiewicz) [reply packages/ui-components/src/components/ContentHeading/ContentHeading.stories.tsx:8]: Done

File: packages/ui-components/src/deprecated_js/Button/button.scss (hunk 4 above)
Comment (gjaskiewicz) [reply packages/ui-components/src/deprecated_js/Button/button.scss:8]: Done

File: packages/ui-components/src/components/Button/Button.test.tsx (hunk 2 above)
Comment (gjaskiewicz) [reply packages/ui-components/src/components/Button/Button.test.tsx:40]: Done

File: packages/ui-components/src/components/Button/Button.test.tsx (hunk 1 above)
Comment (gjaskiewicz) [reply packages/ui-components/src/components/Button/Button.test.tsx:47]: Done

File: packages/ui-components/src/components/ContentHeading/ContentHeading.stories.tsx (hunk 3 above)
Comment (andypf) [inline packages/ui-components/src/components/ContentHeading/ContentHeading.stories.tsx:8]: do we need this commented out code?

**Review by andypf — CHANGES_REQUESTED** [review]

File: packages/ui-components/src/components/ContentHeading/ContentHeading.stories.tsx (hunk 3 above)
Comment (gjaskiewicz) [reply packages/ui-components/src/components/ContentHeading/ContentHeading.stories.tsx]: Done

**Review by andypf — APPROVED** [review]
👍

**Review by edda — APPROVED** [review]

**Pull request merged**